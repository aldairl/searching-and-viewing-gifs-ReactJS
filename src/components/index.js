export * from "./AddCategry";
export * from "./GifGrid";
export * from "./GifItem";
