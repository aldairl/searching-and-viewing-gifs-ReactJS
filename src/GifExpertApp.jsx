import React, { useState } from "react";
import { AddCategry, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categries, setCategries] = useState(["One Punch", "Dragon Ball Z"]);

  const onAddCategory = (newCategory) => {
    if (categries.includes(newCategory)) return;

    setCategries([newCategory, ...categries]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategry onNewCategory={onAddCategory} />

      {categries.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
