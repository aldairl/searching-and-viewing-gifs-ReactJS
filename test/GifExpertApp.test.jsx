import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Test in <GifExpertApp.jsx/>", () => {
  test("should match the snapshop", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });
});
