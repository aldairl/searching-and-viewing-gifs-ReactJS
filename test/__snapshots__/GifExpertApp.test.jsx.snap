// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifExpertApp.jsx/> should match the snapshop 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
     
    One Punch
     
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
     
    Dragon Ball Z
     
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
